File: Lecture3/GetSelection/GetSelection.cpp
#include <maya/MSimple.h>
#include <maya/MGlobal.h>
#include <maya/MString.h>
#include <maya/MDagPath.h>
#include <maya/MFnDagNode.h>
#include <maya/MSelectionList.h>
#include <maya/MIOStream.h>

// This is a macro to create a simple command
// the compiler expands it to a bunch of code

DeclareSimpleCommand( GetSelection , "NCCA", "Any");

MStatus GetSelection::doIt( const MArgList& )
{

  MDagPath node;
  MObject component;
  MSelectionList list;
  MFnDagNode nodeFn;
  // get the currently selected objects
  MGlobal::getActiveSelectionList( list );
  // get the size of the list
  uint size=list.length();
  for (uint  i=0; i< size; ++i )
  {
    // get the DAG path for this object and place the node and component values
    list.getDagPath( i, node, component );
    // attach the found node to the DagNode function set
    nodeFn.setObject( node );
    // now use this to get the object name and display
    MGlobal::displayInfo("Object "+nodeFn.name()+ " is selected");
  }
  return MS::kSuccess;
}

File: Lecture3/GetSelection/get_selection.py
import maya.api.OpenMaya as OM

node_fn = OM.MFnDagNode()
list = OM.MGlobal.getActiveSelectionList()
for i in range(0, list.length()):
    path = list.getDagPath(i)
    node_fn.setObject(path)
    print(f"Object {node_fn.name()} is selected")


dagIt = OM.MItDag(OM.MItDag.kBreadthFirst)
while not dagIt.isDone():
    node = dagIt.currentItem()
    node_fn.setObject(node)
    if node.hasFn(OM.MFn.kNamedObject):
        print(f"Object {node_fn.name()} type {node_fn.typeName}")
        m = node_fn.transformationMatrix()
        print(m)
    dagIt.next()
